api/test/generate-data/detect_model_changes.test.js

```diff
@@ -41,8 +41,12 @@ function filterRelevantFolders(apiFolderJsonObj) {
               break;
             case "test":
                 apiSubfolder.children.forEach(function(testSubfolder) {
-                    if ("factories" == testSubfolder.name) {
-                        factories = testSubfolder.children;
+                    if ("generate-data" == testSubfolder.name) {
```

Comment:
A more functional ES6 style approach could be: 

`factories = apiSubfolder.children.forEach((apiSubfolderChild) => `
`	apiSubfolderChild.name == "generate-data" && apiSubfolderChild.children.forEach((generateDataSubfolderChild) =>`
`		generateDataSubfolderChild.name == "factories" && generateDataSubfolderChild.children));`

Although this might look confusing depending on this is formatted as a comment, it does describe the intent of the code immediately (to set a factories object).

In writing this, I'm a bit confused as to where "factories" is being consumed, because it doesn't appear to be used in the foreach function itself, so then would get set to the children of the last subfolder only?

Reply:
@ActionAnalytics was the mastermind behind the factories. He should be able to answer this. Also I am not quite sure if we support es6 for this repo. I know we don't in our actual controllers but I am not quite sure about our tests.